src/org/zaproxy/zap/extension/custompages/DefaultCustomPage.java

```diff
@@ -0,0 +1,304 @@
+		super();
+		this.contextId = contextId;
+		this.content = content;
+		this.contentType = CustomPageContentType.getCustomPageContentTypeWithId(contentTypeID);
+		this.regex = regex;
+		this.type = CustomPageType.getCustomPageTypeWithId(typeID);
+		this.setEnabled(enabled);
+	}
+	
+	/**
+	 * Returns the context ID for a Custom Page.
+	 * 
+	 *  @return the context ID {@code int}
+	 */
+	@Override
+	public int getContextId() {
+		return contextId;
+	}
+
+	/**
+	 * Sets the context ID for a Custom Page.
+	 * 
+	 * @param contextId the ID of the context
+	 */
+	@Override
+	public void setContextId(int contextId) {
+		this.contextId = contextId;
+	}
+
+	/**
+	 * Returns the content for a Custom Page.
+	 * 
+	 * @return the content {@code String}
+	 */
+	@Override
+	public String getContent() {
+		return content;
+	}
+
+	/**
+	 * Sets the content for a Custom Page.
+	 * 
+	 * @param content the content being set
+	 */
+	@Override
+	public void setContent(String content) {
+		this.content = content;
+	}
+
+	/**
+	 * Returns the {@link CustomPageContentType} {@code enum} literal for a Custom Page.
+	 * 
+	 * @return the custom page content type {@code CustomPageContentType}
+	 */
+	@Override
+	public CustomPageContentType getContentType() {
+		return contentType;
+	}
+
+	/**
+	 * Sets the {@link CustomPageContentType} {@code enum} literal for a Custom Page.
+	 * 
+	 * @param the {@code CustomPageContentType} being set
+	 */
+	@Override
+	public void setContentType(CustomPageContentType cpct) {
+		this.contentType = cpct;
+	}
+	
+	/**
+	 * Returns {@code true} if the {@link #content} of a Custom Page is a Regex pattern, otherwise {@code false}.
+	 * 
+	 * @return a boolean representing whether or not the {@code content} is a Regex pattern
+	 */
+	@Override
+	public boolean isRegex() {
+		return regex;
+	}
+
+	/**
+	 * Sets a boolean designating whether or not the {@link content} of a Custom Page is a Regex pattern.
+	 * 
+	 * @param regex the state being set
+	 */
+	@Override
+	public void setRegex(boolean regex) {
+		this.regex = regex;
+	}
+
+	/**
+	 * Returns the {@code enum} literal {@link CustomPageType} of a Custom Page.
+	 * 
+	 * @return the custom page type
+	 */
+	@Override
+	public CustomPageType getType() {
+		return type;
+	}
+	
+	/**
+	 * Sets the {@link CustomPageType} {@code enum} literal for a Custom Page.
+	 * 
+	 * @param the custom page type to be set
+	 */
+	@Override
+	public void setType(CustomPageType cpt) {
+		this.type = cpt;
+	}
+	
+	/**
+	 * Determines if a {@code HttpMessage} is a Custom Page of a particular {@code CustomPageType}.
+	 * 
+	 * @param msg the HTTP message to be evaluated
+	 * @param cpt the CustomPageType of the Custom Pages against which the HTTP message should be evaluated
+	 * @return {@code true} if the HTTP message is a Custom Page of the type in question, {@code false} otherwise
+	 */
+	public boolean isCustomPage(HttpMessage msg, CustomPageType cpt) {

```

Comment:
Maybe we could make this a bit simpler (fewer nestings)

```
       public boolean isCustomPage(HttpMessage msg, CustomPageType cpt) {
		if (isEnabled() && getType() == cpt) {
			String value = getContentByType(msg);
			return matchByType(value);
		}
		return false; // Default
	}

	private String getContentByType(HttpMessage msg){
		if(CustomPageContentType.URL.equals(getContentType())){
			return msg.getRequestHeader().getURI().toString();
		} else if(CustomPageContentType.RESPONSE_CONTENT.equals(getContentType())){
			return getHttpMessageAsString(msg);
		}
		throw new RuntimeException("Not Supported");
	}

	private boolean matchByType(String value){
		if (!isRegex() && CustomPageContentType.URL.equals(getContentType())) {
			return value.equals(getContent());
		}
		if (!isRegex()){
			return value.contains(getContent());
		}		
		return isRegexMatch(content, value);
	}

```

Reply:
Done